### PR TITLE
Fix can not catch error mongo

### PR DESCRIPTION
**Pull request by Ohen**
Bug Fix context
* Fixed that promise could not be caught if an error occurred in dns.lookup
* Added `adapter initilize timeout` that merged already in master branch

### CODE REVIEW

File: src/adapters/impl/mongoose-adapter.ts [hunk 1]
```diff
@@ -25,25 +25,24 @@ export default class MongooseAdapter extends AbstractAdapter<mongoose.Connection
...
-        connection.once('open', () => {
-          logger.info(`connected to mongo ${address} with ${util.inspect(connectionOptions, { colors: true })}`);
-          this._adaptee = connection;
-          connection.removeAllListeners();
-          resolve();
-        });
-        connection.once('error', err => {
-          logger.info(`connection error on mongo ${address} with ${util.inspect(connectionOptions, { colors: true })}`);
-          reject(err);
-        });
```
Comment (jiho-kr) [inline src/adapters/impl/mongoose-adapter.ts]: Does timeout throw the error?

I think I can do this.
```
this.dnsLookup(uri).then(address => {
...
}).catch(e => reject(e);
```

File: src/adapters/impl/mongoose-adapter.ts [hunk 2]
```diff
@@ -25,25 +25,24 @@ export default class MongooseAdapter extends AbstractAdapter<mongoose.Connection
    * @override
    */
   public initialize() {
-    return new Promise<void>((resolve, reject) => {
+    return new Promise<void>(async (resolve, reject) => {
```
Comment (heycalmdown) [inline src/adapters/impl/mongoose-adapter.ts]: This - `new Promise` - statement can go to where it is actually used to reduce complexity.

File: src/adapters/impl/mongoose-adapter.ts [hunk 3]
```diff
@@ -25,25 +25,24 @@ export default class MongooseAdapter extends AbstractAdapter<mongoose.Connection
...
-          resolve();
-        });
-        connection.once('error', err => {
-          logger.info(`connection error on mongo ${address} with ${util.inspect(connectionOptions, { colors: true })}`);
-          reject(err);
-        });
+      const address = await this.dnsLookup(uri);
+      logger.info(`connecting to mongo ${address} with ${util.inspect(connectionOptions, { colors: true })}`);
+      const connection = mongoose.createConnection(address, connectionOptions);
+      connection.once('open', () => {
```
Comment (heycalmdown) [inline src/adapters/impl/mongoose-adapter.ts]: Maybe here

File: src/adapters/impl/mongoose-adapter.ts (hunk 2 above)
Comment (Ohen) [reply src/adapters/impl/mongoose-adapter.ts]: I was refactoring to code. to reduce complexity

File: src/adapters/impl/mongoose-adapter.ts (hunk 1 above)
Comment (Ohen) [reply src/adapters/impl/mongoose-adapter.ts]: I understand it. but, the code has been changed to the latest style async function, so we do not have to catch anymore.

File: src/adapters/impl/mongoose-adapter.ts (hunk 2 above)
Comment (jiho-kr) [reply src/adapters/impl/mongoose-adapter.ts]: need `async`?

**Pull request merged**